### PR TITLE
Add support for plugins in vba format.

### PR DESCRIPTION
**Pull request by paulhybryant**
Hi Shougo,

I am attempting to add this in my fork. Do you mind giving some feedback?
Specifically I am not quite sure why line 79 and 80 in vba.vim have no effect.

Also it would be great if you are open to merge this PR to support vba.

Thanks,

### CODE REVIEW

File: autoload/neobundle/types/git.vim [hunk 1]
```diff
@@ -209,7 +209,6 @@ function! s:type.get_revision_remote_command(bundle) "{{{
   return g:neobundle#types#git#command_path
         \ .' ls-remote origin ' . rev
 endfunction"}}}
-
```
Comment (Shougo) [inline autoload/neobundle/types/git.vim]: Please don't change git.vim.

File: autoload/neobundle/types/git.vim [hunk 2]
```diff
@@ -234,7 +233,6 @@ function! s:parse_other_pattern(protocol, path, opts) "{{{
 
   return uri
 endfunction"}}}
-
```
Comment (Shougo) [inline autoload/neobundle/types/git.vim]: Same above.

File: autoload/neobundle/types/vba.vim [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  if executable('curl')
+    let cmd = printf('curl --fail -s -o "%s" "%s";', filename, a:bundle.uri)
+  elseif executable('wget')
+    let cmd = printf('wget -q -O "%s" "%s";', filename, a:bundle.uri)
+  else
+    return 'E: curl or wget command is not available!'
+  endif
+  let cmd .= printf(' vim -c "UseVimball %s" -c "q" %s;', path, filename)
+  " let cmd .= printf(' rm %s;', filename)
+  " let cmd .= printf(' rm %s/.VimballRecord', path)
```
Comment (Shougo) [inline autoload/neobundle/types/vba.vim]: Why these lines are commented out?

File: autoload/neobundle/types/vba.vim [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  let a:bundle.type__filepath = filename
+
+  if executable('curl')
+    let cmd = printf('curl --fail -s -o "%s" "%s";', filename, a:bundle.uri)
+  elseif executable('wget')
+    let cmd = printf('wget -q -O "%s" "%s";', filename, a:bundle.uri)
+  else
+    return 'E: curl or wget command is not available!'
+  endif
+  let cmd .= printf(' vim -c "UseVimball %s" -c "q" %s;', path, filename)
```
Comment (Shougo) [inline autoload/neobundle/types/vba.vim]: You should use `v:progpath` instead of "vim".

File: autoload/neobundle/installer.vim [hunk 5]
```diff
@@ -235,8 +235,16 @@ function! neobundle#installer#get_revision_number(bundle)
 
     let rev = neobundle#util#system(cmd)
 
-    " If rev contains spaces, it is error message
-    return (rev !~ '\s') ? rev : ''
+    if type.name ==# "vba" || type.name ==# "raw"
```
Comment (Shougo) [inline autoload/neobundle/installer.vim]: Please use single quotes instead of double quotes.

File: autoload/neobundle/installer.vim [hunk 6]
```diff
@@ -235,8 +235,16 @@ function! neobundle#installer#get_revision_number(bundle)
...
+      " If rev is ok, the output is the checksum followed by the filename
+      " separated by two spaces.
+      let pat = '\v^[0-9a-f]+  ' . a:bundle.path . '/' .
+            \ fnamemodify(a:bundle.uri, ':t') . '$'
+      return (rev =~# pat) ? matchstr(rev, '\v^[0-9a-f]+') : ''
```
Comment (Shougo) [inline autoload/neobundle/installer.vim]: Please don't use `\v` pattern.

File: autoload/neobundle/installer.vim [hunk 7]
```diff
@@ -235,8 +235,16 @@ function! neobundle#installer#get_revision_number(bundle)
 
     let rev = neobundle#util#system(cmd)
 
-    " If rev contains spaces, it is error message
-    return (rev !~ '\s') ? rev : ''
+    if type.name ==# "vba" || type.name ==# "raw"
+      " If rev is ok, the output is the checksum followed by the filename
+      " separated by two spaces.
+      let pat = '\v^[0-9a-f]+  ' . a:bundle.path . '/' .
```
Comment (Shougo) [inline autoload/neobundle/installer.vim]: Please don't use `\v` pattern.

**Comment by Shougo** [comment]
Important problem:  Please add the documentation for vba type.

File: autoload/neobundle/installer.vim (hunk 5 above)
Comment (paulhybryant) [inline autoload/neobundle/installer.vim]: Fixed

File: autoload/neobundle/types/git.vim (hunk 1 above)
Comment (paulhybryant) [inline autoload/neobundle/types/git.vim]: It is just two extra blank lines in git.vim so I deleted them. Restored git.vim

File: autoload/neobundle/types/vba.vim (hunk 4 above)
Comment (paulhybryant) [inline autoload/neobundle/types/vba.vim]: Fixed.

File: autoload/neobundle/types/vba.vim (hunk 3 above)
Comment (paulhybryant) [inline autoload/neobundle/types/vba.vim]: These two lines does't seem to work, and I am curious why. I don't have much background in how vimproc works so I am wondering whether you have some idea?

File: autoload/neobundle/installer.vim (hunk 6 above)
Comment (paulhybryant) [inline autoload/neobundle/installer.vim]: Same as above comments. Which pattern do you recommend?

Comment (Shougo) [inline autoload/neobundle/installer.vim]: It is `^[0-9a-f]\+`.

Comment (paulhybryant) [inline autoload/neobundle/installer.vim]: Sure. Thanks.

On Sun, Aug 30, 2015 at 5:09 PM, Shougo <notifications@github.com> wrote:

> In autoload/neobundle/installer.vim
> <https://github.com/Shougo/neobundle.vim/pull/411#discussion_r38269917>:
>
> > @@ -235,8 +235,16 @@ function! neobundle#installer#get_revision_number(bundle)
> >
> >      let rev = neobundle#util#system(cmd)
> >
> > -    " If rev contains spaces, it is error message
> > -    return (rev !~ '\s') ? rev : ''
> > +    if type.name ==# "vba" || type.name ==# "raw"
> > +      " If rev is ok, the output is the checksum followed by the filename
> > +      " separated by two spaces.
> > +      let pat = '\v^[0-9a-f]+  ' . a:bundle.path . '/' .
> > +            \ fnamemodify(a:bundle.uri, ':t') . '$'
> > +      return (rev =~# pat) ? matchstr(rev, '\v^[0-9a-f]+') : ''
>
> It is ^[0-9a-f]\+.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Shougo/neobundle.vim/pull/411/files#r38269917>.
>



-- 
With Regards,

Yu Huang

Software Engineer,
Google Inc.

**Comment by Shougo** [comment]
OK.  Thanks.

**Pull request merged**